### PR TITLE
Improve renderSubview/registerSubview docs | closes #104

### LINKED ISSUE #104: Explain the difference between renderSubview and registerSubview
This is a minor issue, but it would be nice to provide a brief explanation about the difference between renderSubview and registerSubview. Basically state that renderSubview is a wrapper around registerSubview with some extra sugar (auto render then register).

If I get some time, I can create a PR, but if someone else can get this done before then, the better.

### PR DESCRIPTION
**Pull request by aaronmccall**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -498,7 +498,13 @@ cleanup : function(){
 
 ### registerSubview `view.registerSubview(viewInstance)`
 
-* viewInstance {Object} Any object with a "remove" method, typically an instantiated view. But doesn't have to be, it can be anything with a remove method. The remove method doesn't have to actually remove itself from the DOM (since the parent view is being removed anyway), it is generally just used for unregistering any handler that it set up.
+* viewInstance {Object} Any object with a "remove" method, typically an instantiated view. But doesn't have to be, it can be anything with a remove method. The remove method doesn't have to actually remove itself from the DOM (since the parent view is being removed anyway), it is generally just used for unregistering any handler that the subview sets up.
+
+This method will:
+
+1. store a reference to the subview for cleanup when `remove()` is called.
+2. sdd a reference to itself at `subview.parent`
```
Comment (cdaringe) [inline README.md]: add*

**Comment by kamilogorek** [comment]
:+1: after typo fix.

**Comment by cdaringe** [comment]
@aaronmccall, can you quick fix this?

**Comment by cdaringe** [comment]
follow up doc commit landing immediately after PR closure

**Pull request merged**